packages/web-components/fast-components/src/checkbox/checkbox.styles.ts

```diff
@@ -103,42 +103,42 @@ export const CheckboxStyles = css`
         border-color: ${neutralFocusBehavior.var};
     }
 
-    :host(.checked) .control {
+    :host([checked]) .control {
         background: ${accentFillRestBehavior.var};
         border: calc(var(--outline-width) * 1px) solid ${accentFillRestBehavior.var};
     }
 
-    :host(.checked:not([disabled])) .control:hover {
+    :host([checked]:not([disabled])) .control:hover {
         background: ${accentFillHoverBehavior.var};
         border: calc(var(--outline-width) * 1px) solid ${accentFillHoverBehavior.var};
     }
 
-    :host(.checked:not([disabled])) .control:active {
+    :host([checked]:not([disabled])) .control:active {
         background: ${accentFillActiveBehavior.var};
         border: calc(var(--outline-width) * 1px) solid ${accentFillActiveBehavior.var};
     }
 
-    :host(.checked:${focusVisible}:not([disabled])) .control {
+    :host([checked]:${focusVisible}:not([disabled])) .control {
         box-shadow: 0 0 0 2px var(--background-color), 0 0 0 4px ${
             neutralFocusBehavior.var
         };
         border-color: transparent;
     }
 
 
-    :host(.disabled) .label,
-    :host(.readonly) .label,
-    :host(.readonly) .control,
-    :host(.disabled) .control {
+    :host([disabled]) .label,
+    :host([readonly]) .label,
+    :host([readonly]) .control,
+    :host([disabled]) .control {
         cursor: ${disabledCursor};
     }
 
-    :host(.checked:not(.indeterminate)) .checked-indicator,
+    :host([checked]:not(.indeterminate)) .checked-indicator,
```

Comment:
should `indeterminate` be an attribute selector?

Reply:
indeterminate is only a property at this point

Reply:
Tada! Thanks @nicholasrice

Reply:
Pardon my ignorance but what does this mean exactly? Should be do something about this in this PR or have a follow-up? I'm thinking how to integrate this with for example Blazor and this could cause some trouble for other integrations. Should we use an attribute for this?

Reply:
Sorry, I should be more clear :)

`indeterminate` is a JavaScript property (AKA an "IDL attribute") only and cannot be represented in HTML as an HTML attribute (AKA a "content attribute"), so there is no attribute selector for you to use to style the indeterminate state - you must (at this point) use the class to style this state. The reason for this is historical; the `Checkbox` web-component behaviorally mirrors the `<input type="checkbox />` component, which does not expose an HTML attribute that sets the `indeterminate` property.

Whether we *should* expose `indeterminate` as an attribute could certainly be up for debate, but I think that would need enough investigation that I wouldn't want to resolve that for this PR. In an ideal world, custom element states would allow us to match the `:indeterminate` pseduo-class, but we don't quite have that capability today.

I don't think any action is necessary here, but this control will still exhibit the behavior that this PR aims to fix. I'll open up an issue to track discussion about what we should do about that.

Reply:
#4084